### ISSUE TITLE
There is no description of support of node.js 10

### ISSUE THREAD

**Issue by takahashi-jiec** [RuntimeTools/appmetrics#571]
**I think that the description of Node.js 10 of README.md is necessary.**

----------------------------------------

### Supported platforms

The Node Application Metrics agent supports the following runtime environments where a Node.js runtime is available:

* **Node.js v4** on:
  * 64-bit or 32-bit Windows (x64 or x86)
  * 64-bit or 32-bit Linux (x64, x86, ppc64, ppc64le, s390, s390x)
  * 64-bit AIX (ppc64)
  * 64-bit macOS (x64)
* **Node.js v6** on all of the above, plus:
  * 64-bit runtime on z/OS (os390)
* **Node.js v8** on:
  * 64-bit Windows (x64)
  * 64-bit Linux (x64, ppc64, ppc64le, s390x)
  * 64-bit AIX (ppc64)
  * 64-bit IBM i (ppc64)
    * Before running `npm install appmetrics`, ensure the environment variable `CC=gcc` is set.
    * Functionality for Memory and CPU stats not fully implemented, currently under construction.
  * 64-bit macOS (x64)

--------------------↓--------------------

### Supported platforms

The Node Application Metrics agent supports the following runtime environments where a Node.js runtime is available:

* **Node.js v4** on:
  * 64-bit or 32-bit Windows (x64 or x86)
  * 64-bit or 32-bit Linux (x64, x86, ppc64, ppc64le, s390, s390x)
  * 64-bit AIX (ppc64)
  * 64-bit macOS (x64)
* **Node.js v6** on all of the above, plus:
  * 64-bit runtime on z/OS (os390)
* **Node.js v8** on:
  * 64-bit Windows (x64)
  * 64-bit Linux (x64, ppc64, ppc64le, s390x)
  * 64-bit AIX (ppc64)
  * 64-bit IBM i (ppc64)
    * Before running `npm install appmetrics`, ensure the environment variable `CC=gcc` is set.
    * Functionality for Memory and CPU stats not fully implemented, currently under construction.
  * 64-bit macOS (x64)
* **Node.js v10** on:
  * 64-bit Windows (x64)
  * 64-bit Linux (x64)

----------------------------------------

**Comment by yuecchen** [comment]
will you support win10?

I got an error while compiling on win10:
```
MSBUILD : error MSB4132: The tools version "2.0" is unrecognized. Available tools versions are "4.0".
gyp ERR! build error
gyp ERR! stack Error: `C:\Windows\Microsoft.NET\Framework\v4.0.30319\msbuild.exe` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onExit (C:\Users\YUECHEN\AppData\Roaming\npm\node_modules\npm\node_modules\node-gyp\lib\build.js:262:23)
gyp ERR! stack     at ChildProcess.emit (events.js:189:13)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:248:12)
gyp ERR! System Windows_NT 10.0.17134
gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "C:\\Users\\YUECHEN\\AppData\\Roaming\\npm\\node_modules\\npm\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild"
gyp ERR! cwd C:\projects\nodejs-internal\opentracing\Nodejs-Data-Collector\ibmapm\node_modules\appmetrics
gyp ERR! node -v v10.15.3
gyp ERR! node-gyp -v v3.8.0
gyp ERR! not ok
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! appmetrics@4.0.1 install: `node extract_all_binaries.js || node-gyp rebuild`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the appmetrics@4.0.1 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\YUECHEN\AppData\Roaming\npm-cache\_logs\2019-04-29T01_39_39_169Z-debug.log
```

**Comment by takahashi-jiec** [comment]
Try the following:
```
node extract_all_binaries.js || node-gyp rebuild
```

**Comment by rwalle61** [comment]
@takahashi-jiec we are checking which versions we support right now and will update the doc when we know. For example, I'm currently nearly finished fixing Appmetrics for Node 12 on mac (see  https://github.com/RuntimeTools/appmetrics/issues/572).

**Comment by takahashi-jiec** [comment]
@rwalle61 Thank you

**Comment by mattcolegate** [comment]
@yuecchen, I've confirmed that Appmetrics installs and `npm test`s sucessfully using Node 10 on Windows 10. I suspect the problem you're having is with your build environment, as Appmetrics on Node 10 now has to build native binaries rather than having them shipped with the package. I used https://github.com/nodejs/node-gyp#option-1 to obtain a working environment. It looks like you already have some Microsoft build tools installed, this issue thread might help: https://github.com/chjj/pty.js/issues/60
When we update the readme with statements of supported levels, we will include a link to the `node-gyp` readme for Windows users to make it obvious they need build tools for native binaries.
